tools/autograd/gen_autograd.py

```diff
@@ -88,32 +88,44 @@ def group_declarations_by_signature():
 
     def get_signature(name, params, call_args):
         # create a mapping of parameter name to parameter type
-        types = dict([param.split(' ')[::-1] for param in params])
+        types = dict([param.split(' ')[::-1] for param in params if param != '*'])
         # if the name in the call is not in the parameter list, assume it's
         # a literal Scalar
         rearranged_types = [types.get(arg, 'Scalar') for arg in call_args]
         return '{}({})'.format(name, ', '.join(rearranged_types))
 
     for deprecated in deprecated_defs:
-        python_signature = deprecated['name']
-        call_args = split_name_params(deprecated['aten'])[1]
-        name, params = split_name_params(python_signature)
-        signature = get_signature(name, params, call_args)
+        aten_name, call_args = split_name_params(deprecated['aten'])
+        name, params = split_name_params(deprecated['name'])
+        signature = get_signature(aten_name, params, call_args)
 
         for declaration in declarations_by_signature[signature]:
             declaration = copy.deepcopy(declaration)
             declaration['deprecated'] = True
             declaration['call_args'] = call_args
-            if declaration['inplace']:
-                declaration['python_signature'] = python_signature.replace(name, name + '_')
-            else:
-                declaration['python_signature'] = python_signature
-
-            args_by_name = {arg['name']: arg for arg in declaration['arguments']}
-            declaration['arguments'] = []
-            for arg in params:
-                _, arg_name = arg.split(' ')
-                declaration['arguments'].append(args_by_name[arg_name])
+
+            call_arg_to_idx = {arg: i for i, arg in enumerate(call_args)}
+            original_args = declaration['arguments']
+
+            # Re-order the arguments in the deprecated overload, but keep
+            # the original types.
+            arguments = []
+            kwarg_only = False
+            for param in params:
+                if param == '*':
+                    kwarg_only = True
+                    continue
+                _, param_name = param.split(' ')
+                original = original_args[call_arg_to_idx[param_name]]
+                arguments.append({
```

Comment:
nit: is there some nice way you can copy the original_args and make your changes?  (so you don't have to specify the elements to copy).  I know there is in python 3 (3.6?), but don't know otherwise.

Reply:
We could deepcopy it, but I'd rather be specific about what I copy instead of specific about what I update here. I don't want to copy default values from the original declaration, for example.